### ISSUE TITLE
Filtered list does not update to reflect changes on source list items

### ISSUE THREAD

**Issue by DanielStolt** [RolandPheasant/DynamicData#76]
If I run the following code in LINQPad:

```cs
void Main()
{
	var persons = new Person[]
	{
		new Person(39, "Daniel"),
		new Person(49, "OldDude")
	};

	var sourceList = new SourceList<Person>();
	sourceList.AddRange(persons);
	
	var observableList = sourceList
		.Connect()
		.Filter(x => x.Age < 40)
		.AsObservableList();
		
	observableList.Count.Dump("Number of items in observable list before change");
	
	// Change one of the Person objects so that it now qualifies for the filter.
	persons[1].PropertyChanged += (sender, e) => e.PropertyName.Dump("Property changed");
	persons[1].Age = 30;

	observableList.Count.Dump("Number of items in observable list after change");
}

public class Person : INotifyPropertyChanged
{
	public Person(int age, string name)
	{
		mAge = age;
		mName = name;
	}
	
	private int mAge = 39;
	private string mName = "Daniel";

	public int Age
	{
		get => mAge;
		set
		{
			if (value != mAge)
			{
				mAge = value;
				OnPropertyChanged(nameof(Age));
			}
		}
	}

	public string Name
	{
		get => mName;
		set
		{
			if (value != mName)
			{
				mName = value;
				OnPropertyChanged(nameof(Name));
			}
		}
	}

	public event PropertyChangedEventHandler PropertyChanged;

	protected void OnPropertyChanged(string propertyName)
	{
		PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
	}
}
```

I get the following output:

![image](https://cloud.githubusercontent.com/assets/3889090/25626405/b76f8838-2f5f-11e7-851a-f50cdab5c277.png)

What am I doing wrong? How do I get this observable to react to changes to the list items which would cause items to either pass or fail the filter criteria?

**Comment by RolandPheasant** [comment]
By default dynamic data does not filter on property changes, it only filters on add or removes. However if you want to filter on a property, you can use the FilterOnProperty operator.

There are various examples of filtering in the [snippets project](https://github.com/RolandPheasant/DynamicData.Snippets/tree/master/DynamicData.Snippets/Filter)

**Comment by DanielStolt** [comment]
That works great, thanks. Case closed.

**Comment by anjdreas** [comment]
@RolandPheasant I think `FilterOnProperty` is such a common case, it should be added to the README under the [filtering](https://github.com/RolandPheasant/DynamicData#filtering) section.

The web page (https://dynamic-data.org/) could also have a link to the the snippets folder I was unaware of until now:
https://github.com/RolandPheasant/DynamicData.Snippets/tree/master/DynamicData.Snippets

**Comment by RolandPheasant** [comment]
@anjdreas I need to publicise dd functionality more widely but alas I find I suffer from that classic os developer problem of time starvation.

As a result the thing to suffer is always documentation.  Also the problem with docs is they quickly become outdated and it is difficult to keep them in sync with code changes.

That is why I recently created by the snippets project where the idea is I can quickly make examples to answer any community questions. I have been able to do most of those examples with unit tests in an hour or two whereas the docs often takrs a few evenings in my experience.
